lib/src/query.c

```diff
@@ -119,6 +119,11 @@ typedef struct {
   Array(Slice) slices;
 } SymbolTable;
 
+/**
+ * CaptureQuantififers - a data structure holding the quantifiers of pattern captures.
+ */
+typedef Array(TSQuantifier) CaptureQuantifiers;
```

Comment:
Optimization thoughts -

Currently, we allocate and free one of these lists on each sub-pattern of a query that contains a capture anywhere inside of it. And the length of the list is proportional to the numeric value of the largest capture id, so if a query has 50 captures, we'll allocate a 50-element list for any sub pattern *containing* the 50th capture.

For that reason, I wonder if it's worth making this `Array(uint8_t)`, and casting to `TSQuantifier` when reading from the array, so that we'll only use one byte per member. Currently, I think it'll represent `TSQuantifier` as an `int` (so 4 bytes).

It's a micro-optimization, but a pretty easy one to do. Thoughts?

Reply:
That seems like it might be useful, and will be contained to the implementations of the `capture_quantifier*` procedures. I'll implement that change.